sdk/src/main/java/com/mongodb/stitch/android/services/mongodb/MongoClient.java

```diff
@@ -356,19 +251,28 @@ public Void then(@NonNull final Task<List<Object>> task) throws Exception {
          * @param upsert Whether or not to upsert if the query matches no documents.
          * @return A task that can be resolved upon completion of the request.
          */
-        public Task<Void> updateMany(final Document query, final Document update, final boolean upsert) {
-            return _database._client._stitchClient.executePipeline(makeUpdateStage(query, update, upsert, true)).continueWith(new Continuation<List<Object>, Void>() {
+        public Task<Document> updateMany(final Document query,
+                                         final Document update,
+                                         final boolean upsert) {
+            Document doc = new Document(Parameters.QUERY, query);
+            doc.put(Parameters.DATABASE, _database._dbName);
+            doc.put(Parameters.COLLECTION, _collName);
+            doc.put(Parameters.UPDATE, update);
+            doc.put(Parameters.UPSERT, upsert);
+            doc.put(Parameters.MULTI, true);
+
+            return _database._client._stitchClient.executeServiceFunction(
+                    "updateMany", _database._client._service, doc
+            ).continueWith(new Continuation<Object, Document>() {
                 @Override
-                public Void then(@NonNull final Task<List<Object>> task) throws Exception {
+                public Document then(@NonNull Task<Object> task) throws Exception {
                     if (task.isSuccessful()) {
-                        return null;
+                        Object result = task.getResult();
+                        return Document.parse(result.toString());
+                    } else {
+                        Log.e(TAG, "Error while executing function", task.getException());
```

Comment:
Did we not want to leave these error messages to be more specific?

Reply:
The exception should contain the more specific information.